rover_auxiliary/src/DDB/ddb_control.cpp

```diff
@@ -0,0 +1,373 @@
+#include "ddb_control_node.hpp"
+
+int main(int argc, char** argv)
+{
+    rclcpp::init(argc, argv);
+    auto node = std::make_shared<DDBControlNode>();
+    rclcpp::spin(node);
+    rclcpp::shutdown();
+    return 0;
+}
+
+/**
+ * @brief Construct a new DDBControlNode::DDBControlNode object
+ *
+ */
+DDBControlNode::DDBControlNode():
+    Node("ddb_control")
+{
+    _srv_control_bank0 = this->create_service<rover_msgs::srv::DDBControl>(
+        "/rover/auxiliary/ddb_control_bank0",
+        [this](const std::shared_ptr<rover_msgs::srv::DDBControl::Request> request_,
+               std::shared_ptr<rover_msgs::srv::DDBControl::Response> response_)
+        {
+            if (!request_ || !response_)
+            {
+                RCLCPP_FATAL(this->get_logger(), "NULL request or response received.");
+                return;
+            }
+            this->ddbControlBank0(*request_, *response_);
+        });
+
+    _srv_control_bank1 = this->create_service<rover_msgs::srv::DDBControl>(
+        "/rover/auxiliary/ddb_control_bank1",
+        [this](const std::shared_ptr<rover_msgs::srv::DDBControl::Request> request_,
+               std::shared_ptr<rover_msgs::srv::DDBControl::Response> response_)
+        {
+            if (!request_ || !response_)
+            {
+                RCLCPP_FATAL(this->get_logger(), "NULL request or response received.");
+                return;
+            }
+            this->ddbControlBank1(*request_, *response_);
+        });
+}
+
+/**
+ * @brief Decides what to do depending on the user requested command for bank 0
+ *
+ * @param request_
+ * @param response_
+ */
+void DDBControlNode::ddbControlBank0(const rover_msgs::srv::DDBControl::Request& request_,
+                                     rover_msgs::srv::DDBControl::Response& response_)
+{
+    if (request_.channel_id >= MAX_CHANNELS)
+    {
+        RCLCPP_WARN(this->get_logger(), "Invalid channel (0 to 3). Received channel: %d", request_.channel_id);
+        response_.success = false;
+        response_.status = "Invalid channel (0 to 3). Received channel: " + std::to_string(request_.channel_id);
+        return;
```

Comment:
Prefer doing this instead to remove a bit of copy paste

```suggestion
    if (request_.channel_id >= MAX_CHANNELS)
    {
        std::string msg = "Invalid channel, range is [0; " + std::to_string(MAX_CHANNELS - 1UL) + "]. Received channel : " + std::to_string(request_.channel_id);
        
        RCLCPP_WARN(this->get_logger(), msg);
        response_.success = false;
        response_.status = msg;
        return;
```